File: mobile/src/components/StatusPill.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { colors, radii } from '../theme/colors';

const STATUS_STYLES = {
  success: { bg: colors.successSoft, text: colors.success },
  warning: { bg: colors.warningSoft, text: colors.warning },
  info: { bg: colors.infoSoft, text: colors.info },
  danger: { bg: colors.dangerSoft, text: colors.danger },
  violet: { bg: colors.violetSoft, text: colors.violet },
  neutral: { bg: '#EFEDE8', text: '#555' },
};

export default function StatusPill({ label, tone = 'success', checked = true, style }) {
  const s = STATUS_STYLES[tone] || STATUS_STYLES.success;
  return (
    <View style={[styles.pill, { backgroundColor: s.bg }, style]}>
      {checked && <Text style={[styles.check, { color: s.text }]}>✓</Text>}
      <Text style={[styles.label, { color: s.text }]}>{label}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  pill: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 5,
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: radii.full,
    alignSelf: 'flex-start',
  },
  check: { fontSize: 11, fontWeight: '700' },
  label: { fontSize: 12, fontWeight: '600' },
});

File: mobile_flutter/lib/widgets/status_pill.dart
import 'package:flutter/material.dart';
import '../theme/colors.dart';

enum StatusTone { success, warning, info, danger, violet, neutral }

class StatusPill extends StatelessWidget {
  final String label;
  final StatusTone tone;
  final bool checked;

  const StatusPill({
    super.key,
    required this.label,
    this.tone = StatusTone.success,
    this.checked = true,
  });

  ({Color bg, Color text}) _colors() {
    switch (tone) {
      case StatusTone.success:
        return (bg: AppColors.successSoft, text: AppColors.success);
      case StatusTone.warning:
        return (bg: AppColors.warningSoft, text: AppColors.warning);
      case StatusTone.info:
        return (bg: AppColors.infoSoft, text: AppColors.info);
      case StatusTone.danger:
        return (bg: AppColors.dangerSoft, text: AppColors.danger);
      case StatusTone.violet:
        return (bg: AppColors.violetSoft, text: AppColors.violet);
      case StatusTone.neutral:
        return (bg: const Color(0xFFEFEDE8), text: const Color(0xFF555555));
    }
  }

  @override
  Widget build(BuildContext context) {
    final c = _colors();
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 5),
      decoration: BoxDecoration(
        color: c.bg,
        borderRadius: BorderRadius.circular(AppRadii.full),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          if (checked) ...[
            Text('✓', style: TextStyle(color: c.text, fontSize: 11, fontWeight: FontWeight.w700)),
            const SizedBox(width: 5),
          ],
          Text(
            label,
            style: TextStyle(color: c.text, fontSize: 12, fontWeight: FontWeight.w600),
          ),
        ],
      ),
    );
  }
}
